### PR TITLE
feat: store tinc plugin data in boltdb

### PR DESCRIPTION
**Pull request by antmat**
@oopcode @sshaman1101 @3Hren PTAL

### CODE REVIEW

File: insonmnia/miner/network/tinc_driver.go [hunk 1]
```diff
@@ -69,9 +90,58 @@ type TincNetworkState struct {
...
+			delErr := t.storage.Delete("state")
+			if delErr != nil {
+				t.logger.Errorf("could not cleanup storage for tinc network - %s", delErr)
+			}
+		}
+	}()
+
+	exists, err := t.storage.Exists("state")
+	if err != nil || !exists {
+		return err
```
Comment (sshaman1101) [inline insonmnia/miner/network/tinc_driver.go]: If err == nil && exists == false nil will be returned, but no state was loaded. The caller wouldn't be notified of an error.

File: insonmnia/miner/network/tinc_driver.go [hunk 2]
```diff
@@ -69,9 +90,58 @@ type TincNetworkState struct {
...
+	}
+
+	data, err := t.storage.Get("state")
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(data.Value, t)
+	for _, n := range t.Networks {
+		n.cli = t.cli
```
Comment (sshaman1101) [inline insonmnia/miner/network/tinc_driver.go]: What will be in `t.cli` if unmarshalling is failed? Looks like you can alter the structure internals without error checking.

File: insonmnia/miner/network/tinc_driver.go [hunk 3]
```diff
@@ -69,9 +90,58 @@ type TincNetworkState struct {
...
+		n.logger = t.logger.With("source", "tinc/network/"+n.ID, "container", n.TincContainerID)
+	}
+	return err
+}
+
+func (t *TincNetworkState) sync() error {
+	var err error
+	defer func() {
+		if err != nil {
+			t.logger.Errorf("could not load sync network state - %s; erasing key", err)
```
Comment (sshaman1101) [inline insonmnia/miner/network/tinc_driver.go]: erasing key?

File: insonmnia/miner/network/tinc_driver.go (hunk 1 above)
Comment (antmat) [reply insonmnia/miner/network/tinc_driver.go]: it's ok for us. No data - fresh start

File: insonmnia/miner/network/tinc_driver.go (hunk 2 above)
Comment (antmat) [reply insonmnia/miner/network/tinc_driver.go]: In general that does not matter, but you are right - that would be useless

**Pull request merged**